src/python/zquantum/qcbm/cost_function.py

```diff
@@ -7,73 +8,37 @@
 )
 from zquantum.core.circuit import build_ansatz_circuit
 from zquantum.core.utils import ValueEstimate
+from zquantum.core.gradients import finite_differences_gradient
 from typing import Union, Callable
 import numpy as np
 
 
-class QCBMCostFunction(AnsatzBasedCostFunction):
-    """ Cost function used for evaluating QCBM.
+def QCBMCostFunction(
```

Comment:
If we want to follow Python's naming convention this should be named `qcbm_cost_function` or `QCBM_cost_function`. I recommend changing that unless it would cause some sever compatibility issues.

Reply:
It will cause some backwards compatibility issues on the workflow level if we change the name - would you suggest making it a class instead then?

Reply:
Actually, thinking more on it, it might not cause compatibility issues. However, I'd rather make that change when things are more settled with the workflow-v1 migration